### PR TITLE
Fix the audio task order for precision

### PR DESCRIPTION
**Pull request by khumairraj**
This puts audio operations in order, i.e first convert to Wav format and then do the postprocessing of trimming and padding. This ensures operations are precise since they are done in wav format audio

### CODE REVIEW

File: hearpreprocess/pipeline.py [hunk 1]
```diff
@@ -793,10 +793,14 @@ def run(self):
         self.mark_complete()
 
 
-class MonoWavTrimSubcorpus(MetadataTask):
+class MonoWavSubCorpus(MetadataTask):
```
Comment (turian) [inline hearpreprocess/pipeline.py:796]: Please respect the capitalization convention

File: hearpreprocess/pipeline.py [hunk 2]
```diff
@@ -793,10 +793,14 @@ def run(self):
...
     """
-    Converts the file to mono, changes to wav encoding,
-    trims and pads the audio to be same length
+    Converts the files to wav and mono encoding.
+
+    This task ensures that the audio is converted to an uncompressed format
+    so that any downstream operation on the audio
+    results in precise outputs(like trimming and padding)
```
Comment (turian) [inline hearpreprocess/pipeline.py:802]: whitespace missing

File: hearpreprocess/pipeline.py (hunk 1 above)
Comment (turian) [reply hearpreprocess/pipeline.py:796]: Update the README with the new DAG

File: hearpreprocess/util/audio.py [hunk 3]
```diff
@@ -13,85 +13,90 @@
...
+    except ffmpeg.Error as e:
+        print(
+            "Please check the console output for ffmpeg to debug the "
+            "error in mono wav: ",
+            f"Error: {e}",
+        )
+        raise
+
+
+def trim_pad_wav(in_file: str, out_file: str, out_dur: float) -> None:
```
Comment (turian) [inline hearpreprocess/util/audio.py:34]: Use duration, not out_dir.

File: hearpreprocess/util/audio.py [hunk 4]
```diff
@@ -13,85 +13,90 @@
...
 
 
-def mono_wav_and_fix_duration(in_file: str, out_file: str, duration: float) -> None:
+def mono_wav(in_file: str, out_file: str) -> None:
+    """converts the audio to wav format with mono stream"""
+    try:
+        _ = (
+            ffmpeg.input(in_file)
+            .audio.output(out_file, f="wav", acodec="pcm_f32le", ac=1)
+            .overwrite_output()
```
Comment (turian) [inline hearpreprocess/util/audio.py:22]: Please remove ALL instances of `overwrite_output` in this file.

Please
```
assert not os.path.exists(out_file)
```
for every ffmpeg operation in this file.

The file already being there should be an error.

File: hearpreprocess/util/audio.py [hunk 5]
```diff
@@ -13,85 +13,90 @@
...
+    """
+    # If the audio is of the desired sample rate
+    # move to the else part where we will just create a symlink
+    if get_audio_stats(in_file)["sample_rate"] != out_sr:
         try:
             _ = (
                 ffmpeg.input(in_file)
-                .filter("aresample", resampler="soxr")
+                # Use SoX high quality mode
+                # .filter("aresample", resampler="soxr")
```
Comment (turian) [inline hearpreprocess/util/audio.py:81]: Why? do you disable this? We need sox high quality resampling!

File: hearpreprocess/util/audio.py [hunk 6]
```diff
@@ -13,86 +13,89 @@
...
-def mono_wav_and_fix_duration(in_file: str, out_file: str, duration: float) -> None:
+def mono_wav(in_file: str, out_file: str) -> None:
+    """converts the audio to wav format with mono stream"""
+    assert not Path(out_file).exists(), "File already exists"
+    try:
+        _ = (
+            ffmpeg.input(in_file)
+            .audio.output(out_file, f="wav", acodec="pcm_f32le", ac=1)
+            .run(quiet=True)
+        )
```
Comment (turian) [inline hearpreprocess/util/audio.py:24]: Let's check the out file exists and ffprobe can read it?

**Review by turian — APPROVED** [review]
Looks nice! See my one comment

**Comment by turian** [comment]
@khumairraj have you run this on a corpus with variable length audio files that aren't wavs? one of the secret tasks?

**Comment by khumairraj** [comment]
Yes. I have tested it on task with variable length and it is working as expected.

**Review by turian — APPROVED** [review]

**Pull request merged**